src/screencastwidget.c

```diff
@@ -38,18 +39,62 @@ struct _ScreenCastWidget
 {
   GtkBox parent;
 
+  GtkWidget *source_type_switcher;
+  GtkWidget *window_selection;
+  GtkWidget *monitor_selection;
+
   GtkWidget *monitor_heading;
   GtkWidget *monitor_list;
 
+  GtkWidget *window_heading;
+  GtkWidget *window_list;
+
   DisplayStateTracker *display_state_tracker;
   gulong monitors_changed_handler_id;
   guint selection_changed_timeout_id;
 };
 
 static GQuark quark_monitor_widget_data;
+static GQuark quark_window_widget_data;
 
 G_DEFINE_TYPE (ScreenCastWidget, screen_cast_widget, GTK_TYPE_BOX)
 
+static GtkWidget *
+create_window_widget (Window *window)
+{
+  GtkWidget *window_widget;
+  GtkWidget *window_label;
+  GtkWidget *window_image;
+  GIcon *icon = NULL;
+
+  window_widget = gtk_box_new (GTK_ORIENTATION_HORIZONTAL, 12);
+  gtk_widget_set_margin_start (window_widget, 12);
+  gtk_widget_set_margin_end (window_widget, 12);
+  g_autoptr(GDesktopAppInfo) info = g_desktop_app_info_new (window_get_app_id (window));
```

Comment:
Add `g_autoptr(GDesktopAppInfo) info = NULL` up with the declarations, then do just `info = g_desktop_app_info_new (window_get_app_id (window));` here.

Reply:
Done